### PR TITLE
[codegen/python] Support `<Resource>Args` classes

### PR DESCRIPTION
**Pull request by justinvp**
Add support for creating instances of resources in Python using a `<Resource>Args` class. This capability aligns with how args are passed to resources in all the other language SDKs and the separate object bag allows the properties to be manipulated/validated/passed-around before creating the resource.

Part of #5146

### CODE REVIEW

File: pkg/codegen/python/gen.go [hunk 1]
```diff
@@ -965,34 +982,75 @@ func (mod *modContext) genResource(res *schema.Resource) (string, error) {
...
 
-	// Old versions of TFGen emitted parameters named __name__ and __opts__. In order to preserve backwards
-	// compatibility, we still emit them, but we don't emit documentation for them.
-	fmt.Fprintf(w, ",\n                 __props__=None")
-	fmt.Fprintf(w, ",\n                 __name__=None")
-	fmt.Fprintf(w, ",\n                 __opts__=None):\n")
-	mod.genInitDocstring(w, res)
+	// Now generate __init__ overloads along with an implementation...
+
+	// First, generate an __init__ overload that accepts the resource's inputs from the args class.
```
Comment (justinvp) [inline pkg/codegen/python/gen.go:1003]: Do we have a preference for which overload shows up first in code completion lists? VS Code shows the overloads in the order they're declared (at least in this case).

Screen Shot 2021-03-15 at 12 59 03 PM

Comment (leezen) [reply pkg/codegen/python/gen.go:1003]: I'd propose having the keyword-based style first ahead of the args-based given that's what people are currently using. We could choose to flip this order for 3.0 to encourage the args-based as the 'default'

Comment (justinvp) [reply pkg/codegen/python/gen.go]: SGTM. I've pushed a commit that does that.

Screen Shot 2021-03-16 at 6 22 18 AM

Screen Shot 2021-03-16 at 6 22 41 AM

**Review by komalali — APPROVED** [review]
LGTM. Do we know what impact this will have on docs and how we will handle presenting the overloads there?

**Comment by komalali** [comment]
Oh, I realized that all the tests here are of the codegen itself. Could we add some tests of this in use?

**Pull request merged**